BAU Make published swagger documentation more human readable

stephencdaly opened the pull request:
Group resources in public API documentation into 3 groups:
"Card Payments"
"Direct Debit"
"Refunds"

> NOTE: I think the resources currently in "Refunds" should go into "Card Payments" as they don't apply to Direct Debit. I'd like any opinions on this.

Add human readable names for the resources, which appear in the left hand navigation bar of the API documentation browser.

Find attached a built version of the API browser
[build.zip](https://github.com/alphagov/pay-publicapi/files/3452883/build.zip)

m-green commented on swagger/swagger.json:
```diff
@@ -7,15 +7,20 @@
   "host" : "publicapi.payments.service.gov.uk",
   "tags" : [ {
-    "name" : "refunds"
+    "name" : "Card Payments"
```
```suggestion
    "name" : "Card payments"
```

m-green commented on swagger/swagger.json:
```diff
@@ -7,15 +7,20 @@
+  }, {
+    "name" : "Direct Debit"
+  }, {
+    "name" : "Refunds"
```
```suggestion
    "name" : "Refunding payments"
```

m-green commented on swagger/swagger.json:
```diff
@@ -211,9 +218,10 @@
         "summary" : "Search Direct Debit payments",
         "description" : "Search Direct Debit payments by reference, state, mandate id, and 'from' and 'to' dates. The Authorisation token needs to be specified in the 'Authorization' header as 'Authorization: Bearer YOUR_API_KEY_HERE'",
-        "operationId" : "searchDirectDebitPayments",
+        "operationId" : "Search direct debit payments",
```
```suggestion
        "operationId" : "Search Direct Debit payments",
```

m-green commented on swagger/swagger.json:
```diff
@@ -302,9 +310,10 @@
         "summary" : "Create new Direct Debit payment",
         "description" : "Create a new Direct Debit payment for the account associated to the Authorisation token. The Authorisation token needs to be specified in the 'authorization' header as 'authorization: Bearer YOUR_API_KEY_HERE'",
-        "operationId" : "createDirectDebitPayment",
+        "operationId" : "Collect a direct debit payment",
```
```suggestion
        "operationId" : "Collect a Direct Debit payment",
```

m-green commented on swagger/swagger.json:
```diff
@@ -358,9 +367,10 @@
         "summary" : "Find direct debit payment by ID",
         "description" : "Return information about the direct debit payment. The Authorisation token needs to be specified in the 'Authorization' header as 'Authorization: Bearer YOUR_API_KEY_HERE'",
-        "operationId" : "getDirectDebitPayment",
+        "operationId" : "Get a direct debit payment",
```
```suggestion
        "operationId" : "Get a Direct Debit payment",
```

m-green commented on swagger/swagger.json:
```diff
@@ -511,9 +522,10 @@
         } ]
       },
       "post" : {
+        "tags" : [ "Card Payments" ],
```
```suggestion
        "tags" : [ "Card payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -405,9 +415,10 @@
     },
     "/v1/payments" : {
       "get" : {
+        "tags" : [ "Card Payments" ],
```
```suggestion
        "tags" : [ "Card payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -567,9 +579,10 @@
     },
     "/v1/payments/{paymentId}" : {
       "get" : {
+        "tags" : [ "Card Payments" ],
```
```suggestion
        "tags" : [ "Card payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -615,9 +628,10 @@
     },
     "/v1/payments/{paymentId}/cancel" : {
       "post" : {
+        "tags" : [ "Card Payments" ],
```
```suggestion
        "tags" : [ "Card payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -672,9 +686,10 @@
     },
     "/v1/payments/{paymentId}/capture" : {
       "post" : {
+        "tags" : [ "Card Payments" ],
```
```suggestion
        "tags" : [ "Card payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -729,9 +744,10 @@
     },
     "/v1/payments/{paymentId}/events" : {
       "get" : {
+        "tags" : [ "Card Payments" ],
```
```suggestion
        "tags" : [ "Card payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -777,10 +793,10 @@
     "/v1/payments/{paymentId}/refunds" : {
       "get" : {
-        "tags" : [ "refunds" ],
+        "tags" : [ "Refunds" ],
```
```suggestion
        "tags" : [ "Refunding payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -822,10 +838,10 @@
       },
       "post" : {
-        "tags" : [ "refunds" ],
+        "tags" : [ "Refunds" ],
```
```suggestion
        "tags" : [ "Refunding payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -885,10 +901,10 @@
     "/v1/payments/{paymentId}/refunds/{refundId}" : {
       "get" : {
-        "tags" : [ "refunds" ],
+        "tags" : [ "Refunds" ],
```
```suggestion
        "tags" : [ "Refunding payments" ],
```

m-green commented on swagger/swagger.json:
```diff
@@ -937,10 +953,10 @@
     "/v1/refunds" : {
       "get" : {
-        "tags" : [ "refunds" ],
+        "tags" : [ "Refunds" ],
```
```suggestion
        "tags" : [ "Refunding payments" ],
```

stephencdaly commented:
@m-green Thanks for the comments, I've updated to use the suggestions.

However, instead of "Refunding payments", I renamed the tag to "Refunding card payments" as these APIs can only be used for card payments. Is this sensible? If not I can change back.

m-green commented on swagger/swagger.json:
```diff
@@ -358,9 +367,10 @@
     "/v1/directdebit/payments/{paymentId}" : {
       "get" : {
+        "tags" : [ "Direct Debit" ],
         "summary" : "Find direct debit payment by ID",
```
Sorry, think I missed this one last time!

```suggestion
        "summary" : "Find Direct Debit payment by ID",
```

stephencdaly replied:
Thanks, I've fixed

m-green commented on swagger/swagger.json:
```diff
@@ -358,9 +367,10 @@
     "/v1/directdebit/payments/{paymentId}" : {
       "get" : {
+        "tags" : [ "Direct Debit" ],
         "summary" : "Find direct debit payment by ID",
```
```suggestion
        "summary" : "Find Direct Debit payment by ID",
```

m-green commented on src/main/java/uk/gov/pay/api/model/RefundError.java:
```diff
@@ -15,7 +15,7 @@
 
         SEARCH_REFUNDS_VALIDATION_ERROR("P1101", "Invalid parameters: %s. See Public API documentation for the correct data formats"),
-        SEARCH_REFUNDS_DIRECT_DEBIT_ERROR("P1102","Searching all refunds are not currently supported for direct debit accounts."),
+        SEARCH_REFUNDS_DIRECT_DEBIT_ERROR("P1102","Searching all refunds are not currently supported for Direct Debit accounts."),
```
```suggestion
        SEARCH_REFUNDS_DIRECT_DEBIT_ERROR("P1102","Searching all refunds is not currently supported for Direct Debit accounts."),
```
